pkg/controller/importconfig/cluster_info.go

```diff
@@ -4,297 +4,82 @@
 package importconfig
 
 import (
-	"bytes"
 	"context"
-	"crypto/tls"
-	"crypto/x509"
-	"encoding/base64"
-	"errors"
 	"fmt"
-	"net/url"
-	"os"
 	"reflect"
-	"strings"
 	"time"
 
+	"github.com/stolostron/managedcluster-import-controller/pkg/bootstrap"
 	"github.com/stolostron/managedcluster-import-controller/pkg/constants"
 	"github.com/stolostron/managedcluster-import-controller/pkg/helpers"
-	"github.com/stolostron/managedcluster-import-controller/pkg/helpers/imageregistry"
-	clusterv1 "open-cluster-management.io/api/cluster/v1"
-
-	ocinfrav1 "github.com/openshift/api/config/v1"
 
 	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/types"
-	"k8s.io/apiserver/pkg/storage/names"
-	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/tools/clientcmd"
-	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
-	clientcmdlatest "k8s.io/client-go/tools/clientcmd/api/latest"
-	"k8s.io/utils/pointer"
-
-	authv1 "k8s.io/api/authentication/v1"
-	"sigs.k8s.io/controller-runtime/pkg/client"
+	"k8s.io/klog/v2"
 )
 
-// getBootstrapToken lists the secrets from the managed cluster namespace to look for the managed cluster
-// bootstrap token firstly (compatibility with the ocp that version is less than 4.11), if there is no
-// token found, uses tokenrequest to request token.
-func getBootstrapToken(ctx context.Context, kubeClient kubernetes.Interface,
-	saName, secretNamespace string, tokenExpirationSeconds int64) ([]byte, []byte, error) {
-	secrets, err := kubeClient.CoreV1().Secrets(secretNamespace).List(ctx, metav1.ListOptions{})
-	if err != nil {
-		return nil, nil, err
-	}
-
-	for _, secret := range secrets.Items {
-		if secret.Type != corev1.SecretTypeServiceAccountToken {
-			continue
-		}
+const defaultKlusterletNamespace = "open-cluster-management-agent"
 
-		// refer definitions:
-		// https://github.com/kubernetes/kubernetes/blob/v1.11.0/pkg/controller/serviceaccount/tokens_controller.go#L386
-		// https://github.com/kubernetes/kubernetes/blob/v1.11.0/staging/src/k8s.io/apiserver/pkg/storage/names/generate.go#L49
-		prefix := saName + "-token-"
-		if len(prefix) > names.MaxGeneratedNameLength {
-			prefix = prefix[:names.MaxGeneratedNameLength]
-		}
-
-		if !strings.HasPrefix(secret.Name, prefix) {
-			continue
-		}
-
-		token, ok := secret.Data["token"]
-		if !ok {
-			continue
-		}
-
-		if len(token) == 0 {
-			continue
-		}
-
-		return token, nil, nil
+// getBootstrapKubeConfigDataFromImportSecret if the kubeconfigData it returns is nil, it means
+func getBootstrapKubeConfigDataFromImportSecret(ctx context.Context, clientHolder *helpers.ClientHolder, clusterName string) ([]byte, []byte, error) {
+	// get
+	importSecret, err := getImportSecret(ctx, clientHolder, clusterName)
+	if apierrors.IsNotFound(err) {
+		// recreate bootstrap kubeconfig
```

Comment:
Sure, this is something remain from the previous version where the next line is to create a kubeconfig.

Since in this version, the createkubeconfig part is separated into a independent function, this line can be removed.